### PR TITLE
Remove support for old aot compiler.

### PR DESCRIPTION
**Pull request by meheff**
This is replaced by the new compiler based on cranelift which is much easier to support and generally faster.

### CODE REVIEW

**Comment by meheff** [comment]
@codex review

File: BUILD.bazel [hunk 1]
```diff
@@ -113,9 +113,9 @@ py_test(
         ":xlsynth_sys_dep_probe",
         ":xlsynth_sys_legacy_inputs_probe",
         ":xlsynth_sys_runtime_probe",
+        "@rules_xlsynth_selftest_xls_toolchain",
```
Comment (dank-openai) [inline BUILD.bazel]: Confirmed by running buildifier locally.

`BUILD.bazel` currently fails `buildifier -mode=check`: buildifier wants this `@rules_xlsynth_selftest_xls_toolchain` data label sorted after the two `@rules_xlsynth_selftest_xls_runtime` labels below.

Recommended change: run buildifier on `BUILD.bazel`, or make that one label-order change.

_\[ :robot: posted by Codex on behalf of dank-openai :robot: \]_

Comment (meheff) [reply BUILD.bazel]: fixed.

File: materialize_xls_bundle.py [hunk 2]
```diff
@@ -1360,26 +1156,10 @@ def stage_runtime_payload(repo_root, resolved):
...
-    write_runtime_metadata(
-        repo_root,
-        libxls_dest,
-        xls_aot_runtime_dest,
-        xls_aot_runtime_link_config_dest,
-        xls_aot_runtime_source_repo,
-        runtime_aliases,
-        staged_runtime_files,
-    )
+    write_artifact_config(repo_root, libxls_dest)
```
Comment (dank-openai) [inline materialize_xls_bundle.py:1166]: Optional nit, and fine to ignore if this is more cleanup than it is worth.

The old runtime materialization path could leave `libxls_aot_runtime.a`, `libxls_aot_runtime_link.toml`, and an unpacked `xls_aot_runtime_source/` subpackage under the generated runtime repo. This new path now writes the no-AOT runtime config/metadata, but it does not remove those legacy paths when re-materializing over an existing repo directory.

I do not think clean Bazel builds should usually observe this, because the root `BUILD.bazel` is rewritten and the documented root AOT labels are gone. The edge case is a dirty/reused external repo dir where stale `xls_aot_runtime_source/BUILD.bazel` remains addressable by direct subpackage labels or visible to filesystem-scanning tooling.

Repro: create those three old AOT paths under a temp repo root, call the new `stage_runtime_payload(...)` or `materialize_runtime_surface(...)` with a minimal resolved runtime payload, and observe that the old archive, link config, and `xls_aot_runtime_source/` directory still exist afterward.

If you want to close that edge case, consider deleting those three legacy paths before or during runtime surface materialization, with a small dirty-repo rematerialization test.

_\[ :robot: posted by Codex on behalf of dank-openai :robot: \]_

**Review by dank-openai — APPROVED** [review]

File: materialize_xls_bundle.py (hunk 2 above)
Comment (meheff) [reply materialize_xls_bundle.py:1166]: fixed.
